src/components/customerPortal/PortalCustomerInfos.tsx

```diff
@@ -22,198 +32,209 @@ gql`
       country
       city
       zipcode
+
+      shippingAddress {
+        addressLine1
+        addressLine2
+        city
+        country
+        state
+        zipcode
+      }
     }
   }
 `
 
-interface PortalCustomerInfosProps {
-  translate: Function
+const FieldTitle = ({ title }: { title: string }) => (
+  <p className="text-sm font-normal leading-6 text-grey-600">{title}</p>
+)
+
+const FieldContent = ({ content, children }: { content?: string; children?: React.ReactNode }) => (
+  <p className="break-words text-base font-normal leading-6 text-grey-700">{content || children}</p>
+)
+
+const Field = ({ title, content }: { title: string; content: string }) => (
+  <div className="flex flex-col gap-1">
+    <FieldTitle title={title} />
+    <FieldContent content={content} />
+  </div>
+)
+
+type AddressFieldProps = CustomerAddressInput & {
+  title: string
 }
 
-export const PortalCustomerInfos = memo(({ translate }: PortalCustomerInfosProps) => {
-  const { data, loading } = useGetPortalCustomerInfosQuery()
-  const customerPortalUser = data?.customerPortalUser
+const addressesAreIdentical = ({
+  addressA,
+  addressB,
+}: {
+  addressA?: CustomerAddressInput | null
+  addressB?: CustomerAddressInput | null
+}) =>
+  addressA &&
+  addressB &&
+  addressA.addressLine1 === addressB.addressLine1 &&
+  addressA.addressLine2 === addressB.addressLine2 &&
+  addressA.state === addressB.state &&
+  addressA.country === addressB.country &&
+  addressA.city === addressB.city &&
+  addressA.zipcode === addressB.zipcode
```

Comment:
I'd probably opt for a utility function that is more agnostic about the address keys, in case we decide to add more fields in the future.

Something like this might work—what do you think?

```suggestion
  return (Object.keys(addressA) as (keyof CustomerAddressInput)[]).every(
    (key) => addressA[key] === addressB[key]
  );
```

Reply:
Good point, made it like this, I just had to type out the keys, because I was passing `customerPortalUser` directly.